### DISCUSSION TITLE
Possible to query all consumers with wildcard?

### DISCUSSION THREAD

**Discussion by Evynglais** [nats-io/nats.java#652] [Q&A]
Hi,

I'm attempting to create a NATS admin tool, and I'd like to display all the consumers on one page... but it's pretty slow (approximately 25s+ for under 300 consumers) when using the following query:

```
List<ConsumerInfo> consumerInfo = new ArrayList<>();

for (String stream : jetStreamManagement.getStreamNames()) {
    consumerInfo.addAll(jetStreamManagement.getConsumers(stream));
}

// use consumerInfo to display info on UI
```

Is it possible to query all consumers, rather than per stream? Usable, but 25s delays are obviously not ideal for a UI.

**Comment by scottf** [comment]
Not with the client. If you have monitoring on there might be an http api that returns more information. You could just try to get consumer names instead of the full info, then individually get the consumer info if requested.
